CMakeLists.txt

```diff
@@ -242,6 +242,15 @@ option(OCML_BASIC_ROUNDED_OPERATIONS "Use OCML implementations for devicelib fun
 option(CHIP_MALI_GPU_WORKAROUNDS "Apply work-arounds for avoiding SPIR-V \
 consumption issues in ARM Mali GPU driver." OFF)
 
+# the PowerVR OpenCL implementation for some reason demangles SPIR-V function
+# names, e.g. a SPIRV with a "_Z8testfunc" kernel turned into a cl_program
+# returns "testfunc" in clGetProgramInfo(CL_PROGRAM_KERNEL_NAMES, ...)
+# This is a major issue not only for finding kernels, but also potential
+# name conflicts with function overloads.
+option(CHIP_POWERVR_GPU_WORKAROUNDS "Apply work-arounds for issues in \
```

Comment:
Now realizing that this and the Mali GPU workarounds would have been better to be runtime envs to avoid rebuilding for the GPU. In these platforms the CPU is another valid device. Which makes me realize, you could just detect the need for these workaround from the CL device name, right? That would be much nicer and make the fat binaries runtime portable on the ARM/Mali and RISC-V/PowerVR platforms.

Reply:
^ this should be quite easy to do, right?

Reply:
fixed